README.md

```diff
@@ -0,0 +1,246 @@
+- [🙏 Acknowledgements](#acknowledgements)
+- [📝 License](#license)
+
+<!-- PROJECT DESCRIPTION -->
+
+# 🏡STAY A WHILE BACK-END <a name="about-project"></a>
+Stay A While Backend holds the BackEnd development source rails api of an app that allows users to reserve an apartment or home.
+
+## 🛠 Built With <a name="built-with"></a>
+
+### Tech Stack <a name="tech-stack"></a>
+
+<details>
+  <summary>Server</summary>
+  <ul>
+    <li><a href="https://rubyonrails.org/">Ruby on Rails</a></li>
+  </ul>
+</details>
+
+<details>
+  <summary>Database</summary>
+  <ul>
+    <li><a href="https://www.postgresql.org/">PostgreSQL</a></li>
+  </ul>
+</details>
+
+<details>
+  <summary>UML Diagram</summary>
+  <ul>
+    <li>
+      <img src="./app/assets/images/entity_relationship_diagram.png" alt="diagram"  height="auto" />
+    </li>
+ </ul>
+</details>
+
+<details>
+  <summary>Kanban Board</summary>
+  <ul>
+    <li><a href="https://github.com/users/asharanjith/projects/5/views/2">Project link</a></li>
+    <li>
+      <img src="./app/assets/images/Kanban Board.png" alt="diagram"  height="auto" />
+    </li>
+    
+</li>
+
+ </ul>
+</details>
+
+<!-- Features -->
+
+### Key Features <a name="key-features"></a>
+ - JWT Authorization upon login and register
+ - Create Home Stays and Reservations
+ - Rspec Unit testing
+
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+<!-- LIVE DEMO -->
+## 🚀 Live Demo <a name="live-demo"></a>
+[Stay-A-While](#)
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+## 🚀 Front-End Link <a name="front-end"></a>
+[Stay-A-While](https://github.com/asharanjith/stay-a-while-front-end)
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+<!-- GETTING STARTED -->
+
+## 💻 Getting Started <a name="getting-started"></a>
+
+To get a local copy up and running, follow these steps.
+
+### Prerequisites
+In order to run this project you need:
+
+- To install ruby on your PC. You can read [this](https://rubyinstaller.org/downloads/) documentation on how to do so
+- To install postgresql on your PC. You can read [this](https://www.postgresql.org/) documentation on how to do so
+- Know how to navigate directories or folders at the CLI.
+- Know how to get the URL(https/ssh) of a repository on GitHub.
+- You should have a code editor installed, preferably VSCode
+- In order to run this project you need:
+
+### Setup
+In desired folder or directory in the CLI, run the command:
+
+```sh
+git clone git@github.com:asharanjith/stay-a-while-back-end.git
+```
+
+Navigate into the cloned folder or repository by running the command:
+```sh
+cd stay-a-while-back-end
+```
+
+If VsCode is your default code editor, run:
+```sh
+code .
+```
+
+You are all set up!
+### Install
+
+Install this project with:
+
+```sh
+ bundle install
+``` 
+### Usage
+
+To setup the database, run
+```sh
+rails db:drop db:create db:migrate
+```
+### Run tests
+
+To run tests, run the following command:
+
+```sh
+  rspec spec
+```
```

Comment:
- [ ] When I run the tests according to instructions, this happens:

![image](https://github.com/asharanjith/stay-a-while-back-end/assets/57421823/b2b2da1d-8d44-4478-a6b1-7d3eba4003d4)

## Explanation

It seems that the secret key is missing because it's returning as `nil`.

If this is a bug, kindly fix it so your tests pass.  If it's not, this means your app/tests require special procedures to be run correctly. Kindly add the specific instructions about how to setup the app and tests to make them pass.

_tip:  Is a .env file needed?  Is a special command needed to be run before the tests?  Any other special setup procedures?  Kindly include those instructions on your readme so all your users are able to run your app smoothly._

- [ ] `spec/models/image_spec.rb` is not implemented.  Kindly either implement this test or remove it:

![image](https://github.com/asharanjith/stay-a-while-back-end/assets/57421823/66adec7b-bd2a-433b-a504-30a10e31aff6)

- [ ] `spec/models/requests` has 16 failures and 26 pending:

![image](https://github.com/asharanjith/stay-a-while-back-end/assets/57421823/64be0793-f995-45c6-8e53-6d663233b30c)

Kindly correct those tests and/or eliminate the ones you are not implementing. 👍 

- [ ] Kindly make your tests instructions more specific so all your tests pass.  In order to achieve it, we need to skip the integration tests (the swagger tests) from the command.

You could either do:

```
rspec --exclude-pattern 'spec/integration/*_spec.rb'
```
 
-or-

```
rspec spec/requests
rspec spec/models
rspec spec/routing
```

Kindly update your readme instructions to reflect this.

That way, once fixed, all your tests will pass for all your users. 💪